Framework/Kernel/src/TimeROI.cpp

```diff
@@ -243,6 +266,44 @@ bool TimeROI::valueAtTime(const DateAndTime &time) const {
   }
 }
 
+bool TimeROI::valueAtTime(const std::vector<Types::Core::DateAndTime>::iterator &time) {
+  if (std::distance(m_roi.begin(), time) % 2 == 0)
+    return ROI_USE;
+  else
+    return ROI_IGNORE;
+}
+
+/**
+ * Returns the time supplied if it is in a "use" region, or the minimum of the next higher use region.
```

Comment:
If the time `t` is not in a "use" region, then the time must be in a "mask" region. So for masked time you're looking for the next time in `m_roi` with an associated value of `ROI_USE`?

Reply:
That is correct. This is used in generating `FilteredTimeSeriesProperty::applyFilter()` (of the never ending branch) to support generating time intervals.